src/core/gfx/texture-cache.js

```diff
@@ -28,7 +28,7 @@ class TextureCacheEntry {
   }
 
   get loadingComplete() {
-    return this.loadingCompletePromise && this.loadingCompletePromise.promise;
+    return this.loadingCompletePromise?.promise;
```

Comment:
There's a subtle behavior change here.

In js, a short-circuited `&&` expression will return the first falsey operand, while a short-circuited optional chain will return `undefined`. 

After the promise is resolved:
- the `loadingCompletePromise` is set to `null`
- the original code will return `null`
- the new code will return `undefined`

We should be able to get equivalent behavior with the nullish coalescing operator:
```js
return this.loadingCompletePromise?.promise ?? null;
```